FBRetainCycleDetector/Layout/Classes/Parser/Struct.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "Type.h"
+
+#import <memory>
+#import <string>
+#import <vector>
+
+namespace FB { namespace RetainCycleDetector { namespace Parser {
+  class Struct: public Type {
+  public:
+    const std::string structTypeName;
+    
+    Struct(std::string name,
+           std::string typeEncoding,
+           std::string structTypeName,
+           std::vector<std::shared_ptr<Type>> &typesContainedInStruct)
+    : Type(name, typeEncoding),
+      structTypeName(structTypeName),
+    typesContainedInStruct(std::move(typesContainedInStruct)) {};
```

Comment:
Either this `move` should be removed, or the declared type of `typesContainedInStruct` should be changed to `&&`.